tests/models/test_tpu.py

```diff
@@ -0,0 +1,122 @@
+import os
+from unittest.mock import patch
+
+import pytest
+from pytorch_lightning import Trainer
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.base import EvalModelTemplate
+
+try:
+    import torch_xla
+except ImportError:
+    XLA_AVAILABLE = False
+else:
+    XLA_AVAILABLE = True
+
+
+@pytest.mark.skipif(not XLA_AVAILABLE, reason="test requires TPU machine")
+@pytest.mark.parametrize(['tpu_cores', 'expected_device'], [
+    pytest.param([1], 'xla:1'),
+    pytest.param([8], 'xla:8'),
+])
+def test_single_tpu_core_model(tmpdir, tpu_cores, expected_device):
+    """Test if single TPU core training works"""
+    trainer_options = dict(
+        default_root_dir=tmpdir,
+        progress_bar_refresh_rate=0,
+        max_epochs=1,
+        train_percent_check=0.1,
+        val_percent_check=0.1,
+        tpu_cores=tpu_cores
+    )
+
+    model = EvalModelTemplate()
+    trainer = Trainer(**trainer_options)
+    trainer.fit(model)
+    assert torch_xla._XLAC._xla_get_default_device() == expected_device
+
+
+@pytest.mark.spawn
```

Comment:
recently we moved away from `spawn` for DDP, so do we need it here? 
cc: @williamFalcon

Reply:
I wasn't sure about this and if it's required during TPU training. I will omit it if not required. :]

Reply:
can we just remove the `@pytest.mark.spawn`

Reply:
.spawn is needed for anything multi-process. do not remove or the tests will hang and never finish